transform-plugins/src/main/java/co/cask/hydrator/plugin/XMLParser.java

```diff
@@ -0,0 +1,314 @@
+        if (nodeList.getLength() > 1) {
+          throw new IllegalArgumentException("Cannot specify XPath that are arrays");
+        }
+        node = nodeList.item(0);
+        //Since all columns have nullable schema extracting not nullable type.
+        builder.set(fieldName, parseValues(getValue(node, field.getSchema().getNonNullable().getType(), fieldName),
+                                           field));
+      }
+      emitter.emit(builder.build());
+    } catch (Exception e) {
+      switch (mapErrorProcessing.get(config.processOnError)) {
+        case 2:
+          throw new IllegalStateException("Terminating process in error : " + e.getMessage());
+        case 3:
+          emitter.emitError(new InvalidEntry<>(31, e.getStackTrace()[0].toString() + " : " + e.getMessage(), input));
+          break;
+        default:      //ignore on error (case 1)
+          break;
+      }
+    }
+  }
+
+  /**
+   * Get the node value to be parsed into the required format by parseValues().
+   *
+   * @param node      Node from which the text has to be extracted
+   * @param type      schema type to check if it is a nullable string, in case the xpath evaluates to node with children
+   * @param fieldName field name for which the type is to be evaluated
+   * @return node value as string
+   */
+  private String getValue(Node node, Schema.Type type, String fieldName) {
+    if (node != null) {
+      Node firstChild = node.getFirstChild();
+      //If the xpath evaluates to node which contains child element, the output will be an xml record
+      if (firstChild != null && (firstChild.getNodeType() == Node.ELEMENT_NODE || (firstChild.getNextSibling()
+        != null && (firstChild.getNextSibling().getNodeType() == Node.ELEMENT_NODE)))) {
+        if (!type.equals(Schema.Type.STRING)) {
+          throw new IllegalArgumentException(String.format("The xpath returned node which contains child nodes. " +
+                                                             "Cannot convert %s  to type %s", fieldName, type));
+        } else {
+          return nodeToString(node.cloneNode(true));
+        }
+      } else {
+        return node.getTextContent();
+      }
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Convert node to string to be returned in the output, for cases which contains child elements.
+   *
+   * @param node node to be converted to string
+   * @return converted node as string
+   */
+  private String nodeToString(Node node) {
+    StringWriter stringWriter = new StringWriter();
+    try {
+      Transformer transformer = TransformerFactory.newInstance().newTransformer();
+      transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
+      transformer.setOutputProperty(OutputKeys.INDENT, "no");
+      transformer.transform(new DOMSource(node), new StreamResult(stringWriter));
+    } catch (TransformerException te) {
+      System.out.println("nodeToString Transformer Exception");
```

Comment:
don't write to std out.
use LOG.warn/error/info/debug, as appropriate.
Also, do we want to return empty string in the case that there's a TransformerException? Or should the exception be propagated?